meow/cell.py

```diff
@@ -132,21 +124,43 @@ def create_cells(
             mesh=mesh,
             z_min=z_min,
             z_max=z_max,
-            ez_interfaces=ez_interfaces,
         )
         for mesh, (z_min, z_max) in zip(meshes, zip(z[:-1], z[1:]))
     ]
 
     return cells
 
 
-def _create_material_array(cell, structures, ez_interfaces):
-    m_full = np.zeros_like(cell.mesh.X_full, dtype=np.int_)
+def _create_full_material_array(
+    mesh: Mesh2D,
+    structures: List[Structure2D],
+    materials: Dict[Material, int],
+):
+    m_full = np.zeros_like(mesh.X_full, dtype=np.int_)
+    structures_dict = classify_structures_by_mesh_order_and_material(
```

Comment:
Reminder to self: potential problem with dict ordering

Reply:
python dicts are guarenteed to be ordered since python 3.6

Reply:
Seems I am not up to date then... Sorry